File: Commercial_Controller/Java/commercial_controller/src/app/Battery.java
package app;
import java.util.ArrayList;
//import  java.lang.reflect.Array;

public class Battery{
    
    public int numberOfColumn;
    public int floorNumbersTotal;
    public ArrayList<Column> columnList;
      
    public Battery(int elevatorNumbersTotal, int numberOfColumn, int floorNumbersTotal, int numberOfBasement){
    this.numberOfColumn = numberOfColumn;
    this.floorNumbersTotal = floorNumbersTotal;
    this.columnList = new ArrayList<Column>(); 
    var numberOfElevators = elevatorNumbersTotal / numberOfColumn;
    var numberOfFloors = floorNumbersTotal / (numberOfColumn - 1);

    
//                                  *****GENERATING COLUMNS*****
for (var i = 1; i <= numberOfColumn; i++){
    var col = new Column("Column" + i, numberOfElevators, numberOfFloors);
    columnList.add(col);

//                                  *****GENERATING ELEVATORS FOR EACH COLUMN*****
    for (var j =1; j <= numberOfElevators; j++){
        var elev = new Elevator("Elevator" + i + "." + j, 1, "idle", 1);
        col.elevatorList.add(elev);
        }
    }

for (var i = 0; i < numberOfColumn; i++ ){
    for (var k = 1; k <= numberOfFloors; k++){
        if (i == 0){
            if (k == 1) {
                columnList.get(i).floorList.add(k);
            }
            if (k <= numberOfBasement) {
                columnList.get(i).floorList.add(-k);
            }
        } else if (i == 1){
            columnList.get(i).floorList.add(k);
        } else {
            if (k == 1) {
                columnList.get(i).floorList.add(1);
            }
            columnList.get(i).floorList.add( k+((i-1)*numberOfFloors));


            }
        }
    }
    }
    public void requestElevator(int floorNumber){
        if (floorNumber != 1){
          int requestedFloor = 1;
          this.findColumn(floorNumber, requestedFloor);
      
        } 
  }
      public void assignElevator(int requestedFloor){
        if (requestedFloor != 1){
         int floorNumber = 1;
        this.findColumn(floorNumber, requestedFloor);
      } 
  }
  Column findColumn (int floorNumber, int requestedFloor){
    Column bestCol = this.columnList.get(0);
     int size = this.columnList.size();
    for (var i =0; i < size; i++)
    {
      for (int k : this.columnList.get(i).floorList){
          if (floorNumber == 1 && requestedFloor == k){
              bestCol = this.columnList.get(i);
          } else if (floorNumber == k && requestedFloor == 1){
              bestCol = this.columnList.get(i);
          }
      }
  }
  
    bestCol.findElevator(floorNumber, requestedFloor, bestCol);
  return bestCol;
}
}

File: Commercial_Controller/C#/class/Battery.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Rocket_Elevators_Commercial_Controller.Commercial_Controller.Classes
{
    internal class Battery
    {
        public int ColumnCount;
        public int FloorsCount;
        public List<Column> ColumnList = new List<Column>();
        public Column SelectedColumn = null;

        public Battery(int elevatorsCount, int columnCount, int floorsCount, int basementFloorCount)
        {
            this.ColumnCount = columnCount;
            this.FloorsCount = floorsCount;
            var numberOfElevators = elevatorsCount / this.ColumnCount;
            int numberOfFloors = this.FloorsCount / (this.ColumnCount - 1);

            for (var i = 1; i <= columnCount; i++)
            {
                Column column = new Column("Column " + i);
                this.ColumnList.Add(column);
                    for (var j = 1; j <= numberOfElevators; j++)
                {
                    column.AddElevator(new Elevator("Elevator " + i + "-" + j, 1, "Idle"));
                }
            }

            for (var i = 0; i < columnCount; i++)
            {
                if (i == 0)
                {
                    this.ColumnList[i].FloorsMin = -basementFloorCount;
                    this.ColumnList[i].FloorsMax = 1;
                }
                else
                {
                    this.ColumnList[i].FloorsMin = (i - 1) * numberOfFloors + 1;
                    this.ColumnList[i].FloorsMax = i * numberOfFloors;
                }
            }
        }

        public void RequestElevator(int floorNumber)
        {
            if (floorNumber != 1)
            {
                int requestedFloor = 1;
                this.FindColumn(floorNumber, requestedFloor);

            }
        }
        public void AssignElevator(int requestedFloor)
        {
            if (requestedFloor != 1)
            {
                int floorNumber = 1;
                this.FindColumn(floorNumber, requestedFloor);
            }
        }


        Column FindColumn(int requestedFloor, int destinationFloor)
        {
            if (destinationFloor < 0) this.SelectedColumn = this.ColumnList[0];
            else
            {
                for (var i = 0; i < this.ColumnList.Count; i++)
                {
                    if (requestedFloor == 1 && destinationFloor <= this.ColumnList[i].FloorsMax && destinationFloor >= this.ColumnList[i].FloorsMin) this.SelectedColumn = this.ColumnList[i];
                    else if (requestedFloor <= this.ColumnList[i].FloorsMax && requestedFloor >= this.ColumnList[i].FloorsMin && destinationFloor == 1) this.SelectedColumn = this.ColumnList[i];
                }
            }
            
            if (this.SelectedColumn == null) throw new Exception("Aucune colonne n'a été sélectionné.");
            this.SelectedColumn.FindElevator(requestedFloor, destinationFloor);
            return this.SelectedColumn;
        }
    }
}
